seatunnel-connectors-v2/connector-kafka/src/main/java/org/apache/seatunnel/connectors/seatunnel/kafka/source/KafkaSourceConfig.java

```diff
@@ -0,0 +1,250 @@
+
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.BOOTSTRAP_SERVERS;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.COMMIT_ON_CHECKPOINT;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.CONNECTOR_IDENTITY;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.CONSUMER_GROUP;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.DEBEZIUM_RECORD_INCLUDE_SCHEMA;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.FIELD_DELIMITER;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.FORMAT;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.KAFKA_CONFIG;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.KEY_PARTITION_DISCOVERY_INTERVAL_MILLIS;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.MESSAGE_FORMAT_ERROR_HANDLE_WAY_OPTION;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.PATTERN;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.START_MODE;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.START_MODE_OFFSETS;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.START_MODE_TIMESTAMP;
+import static org.apache.seatunnel.connectors.seatunnel.kafka.config.Config.TOPIC;
+
+public class KafkaSourceConfig implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    @Getter private final ConsumerMetadata metadata;
+
+    @Getter private final DeserializationSchema<SeaTunnelRow> deserializationSchema;
+
+    @Getter private final CatalogTable catalogTable;
+
+    @Getter private final MessageFormatErrorHandleWay messageFormatErrorHandleWay;
+
+    @Getter private final long discoveryIntervalMillis;
+
+    public KafkaSourceConfig(ReadonlyConfig readonlyConfig) {
+        this.metadata = createConsumerMetadata(readonlyConfig);
+        this.discoveryIntervalMillis = readonlyConfig.get(KEY_PARTITION_DISCOVERY_INTERVAL_MILLIS);
+        this.messageFormatErrorHandleWay =
+                readonlyConfig.get(MESSAGE_FORMAT_ERROR_HANDLE_WAY_OPTION);
+        this.catalogTable = createCatalogTable(readonlyConfig);
+        this.deserializationSchema = createDeserializationSchema(catalogTable, readonlyConfig);
+    }
+
+    private ConsumerMetadata createConsumerMetadata(ReadonlyConfig readonlyConfig) {
+        CheckResult result = readonlyConfig.checkAllExists(TOPIC, BOOTSTRAP_SERVERS);
+        if (!result.isSuccess()) {
+            throw new KafkaConnectorException(
+                    SeaTunnelAPIErrorCode.CONFIG_VALIDATION_FAILED,
+                    String.format(
+                            "PluginName: %s, PluginType: %s, Message: %s",
+                            CONNECTOR_IDENTITY, PluginType.SOURCE, result.getMsg()));
+        }
```

Comment:
I think `checkAllExists` is unnecessary. As `ReadonlyConfig`, if it implement `TableSourceFactory`, we will check whether all configs are legal. Please refer https://github.com/apache/seatunnel/blob/aefc135195f5a10e37aec84e0b7c29708342cfcd/seatunnel-api/src/main/java/org/apache/seatunnel/api/table/factory/FactoryUtil.java#L116